cmd/serve/http tests fail with Go 1.27

### Before you start

- [x] I have searched the [forum](https://forum.rclone.org/) and the existing issues for this problem.
- [x] I have tested with the latest beta or stable release and can still reproduce the problem.
- [x] This is a reproducible bug, not a usage question (questions belong on the forum).

### Associated forum post URL

_No response_

### What is the problem you are having with rclone?

Fedora is introducing go-1.27rc to rawhide and rclone builds started to fail.

~~~
github.com/rclone/rclone/cmd/serve/http
2026/07/13 20:56:54 NOTICE: Local file system at /builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/testdata/files: HTTP Server started on [http://127.0.0.1:40121/]
--- FAIL: TestGET (0.09s)
    http_test.go:107: 
        	Error Trace:	/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:107
        	            				/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:289
        	            				/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:294
        	Error:      	Not equal: 
        	            	expected: []string{"PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\b\x00\t\x00one%.txtUT\x05\x00\x01%\xc0n8\xca\xcfKU\xe5\x02\x04\x00\x00\xff\xffPK\a\b\x83\xa7\x83@\v\x00\x00\x00\x05\x00\x00\x00PK\x03\x04\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00three/UT\x05\x00\x01%\xc0n8PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/a.txtUT\x05\x00\x01%\xc0n8*\xc9(JM\xe5\x02\x04\x00\x00\xff\xffPK\a\b\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/b.txtUT\x05\x00\x01%\xc0n8*\xc9(JMM\xe2\x02\x04\x00\x00\xff\xffPK\a\b\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\a\x00\t\x00two.txtUT\x05\x00\x01%\xc0n820426153\xb7\xb0\xe4\x02\x04\x00\x00\xff\xffPK\a\b\xd1\xe5v@\x11\x00\x00\x00\v\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x83\xa7\x83@\v\x00\x00\x00\x05\x00\x00\x00\b\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00one%.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00J\x00\x00\x00three/UT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00w\x00\x00\x00three/a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xc5\x00\x00\x00three/b.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd1\xe5v@\x11\x00\x00\x00\v\x00\x00\x00\a\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x14\x01\x00\x00two.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x05\x00\x05\x00>\x01\x00\x00c\x01\x00\x00\x00\x00"}
        	            	actual  : []string{"PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\b\x00\t\x00one%.txtUT\x05\x00\x01%\xc0n8\x00\x05\x00\xfa\xffone%", "\x03\x00PK\a\b\x83\xa7\x83@\f\x00\x00\x00\x05\x00\x00\x00PK\x03\x04\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00three/UT\x05\x00\x01%\xc0n8PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/a.txtUT\x05\x00\x01%\xc0n8\x00\x06\x00\xf9\xffthree", "\x03\x00PK\a\b\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/b.txtUT\x05\x00\x01%\xc0n8\x00\a\x00\xf8\xffthreeb", "\x03\x00PK\a\b\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\a\x00\t\x00two.txtUT\x05\x00\x01%\xc0n8\x00\v\x00\xf4\xff0123456789", "\x03\x00PK\a\b\xd1\xe5v@\x12\x00\x00\x00\v\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x83\xa7\x83@\f\x00\x00\x00\x05\x00\x00\x00\b\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00one%.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00K\x00\x00\x00three/UT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00x\x00\x00\x00three/a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xc7\x00\x00\x00three/b.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd1\xe5v@\x12\x00\x00\x00\v\x00\x00\x00\a\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x17\x01\x00\x00two.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x05\x00\x05\x00>\x01\x00\x00g\x01\x00\x00\x00\x00"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,7 @@
        	            	-([]string) (len=1) {
        	            	- (string) (len=695) "PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\b\x00\t\x00one%.txtUT\x05\x00\x01%\xc0n8\xca\xcfKU\xe5\x02\x04\x00\x00\xff\xffPK\a\b\x83\xa7\x83@\v\x00\x00\x00\x05\x00\x00\x00PK\x03\x04\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00three/UT\x05\x00\x01%\xc0n8PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/a.txtUT\x05\x00\x01%\xc0n8*\xc9(JM\xe5\x02\x04\x00\x00\xff\xffPK\a\b\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/b.txtUT\x05\x00\x01%\xc0n8*\xc9(JMM\xe2\x02\x04\x00\x00\xff\xffPK\a\b\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\a\x00\t\x00two.txtUT\x05\x00\x01%\xc0n820426153\xb7\xb0\xe4\x02\x04\x00\x00\xff\xffPK\a\b\xd1\xe5v@\x11\x00\x00\x00\v\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x83\xa7\x83@\v\x00\x00\x00\x05\x00\x00\x00\b\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00one%.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00J\x00\x00\x00three/UT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00w\x00\x00\x00three/a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xc5\x00\x00\x00three/b.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd1\xe5v@\x11\x00\x00\x00\v\x00\x00\x00\a\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x14\x01\x00\x00two.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x05\x00\x05\x00>\x01\x00\x00c\x01\x00\x00\x00\x00"
        	            	+([]string) (len=5) {
        	            	+ (string) (len=56) "PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\b\x00\t\x00one%.txtUT\x05\x00\x01%\xc0n8\x00\x05\x00\xfa\xffone%",
        	            	+ (string) (len=123) "\x03\x00PK\a\b\x83\xa7\x83@\f\x00\x00\x00\x05\x00\x00\x00PK\x03\x04\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00three/UT\x05\x00\x01%\xc0n8PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/a.txtUT\x05\x00\x01%\xc0n8\x00\x06\x00\xf9\xffthree",
        	            	+ (string) (len=79) "\x03\x00PK\a\b\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/b.txtUT\x05\x00\x01%\xc0n8\x00\a\x00\xf8\xffthreeb",
        	            	+ (string) (len=79) "\x03\x00PK\a\b\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\a\x00\t\x00two.txtUT\x05\x00\x01%\xc0n8\x00\v\x00\xf4\xff0123456789",
        	            	+ (string) (len=358) "\x03\x00PK\a\b\xd1\xe5v@\x12\x00\x00\x00\v\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x83\xa7\x83@\f\x00\x00\x00\x05\x00\x00\x00\b\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00one%.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00K\x00\x00\x00three/UT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00x\x00\x00\x00three/a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xc7\x00\x00\x00three/b.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd1\xe5v@\x12\x00\x00\x00\v\x00\x00\x00\a\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x17\x01\x00\x00two.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x05\x00\x05\x00>\x01\x00\x00g\x01\x00\x00\x00\x00"
        	            	 }
        	Test:       	TestGET
        	Messages:   	testdata/golden/root.zip
    http_test.go:107: 
        	Error Trace:	/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:107
        	            				/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:289
        	            				/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:294
        	Error:      	Not equal: 
        	            	expected: []string{"PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00a.txtUT\x05\x00\x01%\xc0n8*\xc9(JM\xe5\x02\x04\x00\x00\xff\xffPK\a\b\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00b.txtUT\x05\x00\x01%\xc0n8*\xc9(JMM\xe2\x02\x04\x00\x00\xff\xffPK\a\b\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00H\x00\x00\x00b.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x02\x00\x02\x00x\x00\x00\x00\x91\x00\x00\x00\x00\x00"}
        	            	actual  : []string{"PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00a.txtUT\x05\x00\x01%\xc0n8\x00\x06\x00\xf9\xffthree", "\x03\x00PK\a\b\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00b.txtUT\x05\x00\x01%\xc0n8\x00\a\x00\xf8\xffthreeb", "\x03\x00PK\a\b\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00I\x00\x00\x00b.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x02\x00\x02\x00x\x00\x00\x00\x93\x00\x00\x00\x00\x00"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,5 @@
        	            	-([]string) (len=1) {
        	            	- (string) (len=287) "PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00a.txtUT\x05\x00\x01%\xc0n8*\xc9(JM\xe5\x02\x04\x00\x00\xff\xffPK\a\b\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00b.txtUT\x05\x00\x01%\xc0n8*\xc9(JMM\xe2\x02\x04\x00\x00\xff\xffPK\a\b\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00H\x00\x00\x00b.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x02\x00\x02\x00x\x00\x00\x00\x91\x00\x00\x00\x00\x00"
        	            	+([]string) (len=3) {
        	            	+ (string) (len=54) "PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00a.txtUT\x05\x00\x01%\xc0n8\x00\x06\x00\xf9\xffthree",
        	            	+ (string) (len=73) "\x03\x00PK\a\b\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00b.txtUT\x05\x00\x01%\xc0n8\x00\a\x00\xf8\xffthreeb",
        	            	+ (string) (len=160) "\x03\x00PK\a\b\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00I\x00\x00\x00b.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x02\x00\x02\x00x\x00\x00\x00\x93\x00\x00\x00\x00\x00"
        	            	 }
        	Test:       	TestGET
        	Messages:   	testdata/golden/three.zip
2026/07/13 20:56:54 NOTICE: HTTP Server started on [http://127.0.0.1:43881/]
--- FAIL: TestAuthProxy (4.82s)
    http_test.go:107: 
        	Error Trace:	/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:107
        	            				/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:289
        	            				/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:298
        	Error:      	Not equal: 
        	            	expected: []string{"PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\b\x00\t\x00one%.txtUT\x05\x00\x01%\xc0n8\xca\xcfKU\xe5\x02\x04\x00\x00\xff\xffPK\a\b\x83\xa7\x83@\v\x00\x00\x00\x05\x00\x00\x00PK\x03\x04\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00three/UT\x05\x00\x01%\xc0n8PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/a.txtUT\x05\x00\x01%\xc0n8*\xc9(JM\xe5\x02\x04\x00\x00\xff\xffPK\a\b\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/b.txtUT\x05\x00\x01%\xc0n8*\xc9(JMM\xe2\x02\x04\x00\x00\xff\xffPK\a\b\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\a\x00\t\x00two.txtUT\x05\x00\x01%\xc0n820426153\xb7\xb0\xe4\x02\x04\x00\x00\xff\xffPK\a\b\xd1\xe5v@\x11\x00\x00\x00\v\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x83\xa7\x83@\v\x00\x00\x00\x05\x00\x00\x00\b\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00one%.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00J\x00\x00\x00three/UT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00w\x00\x00\x00three/a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xc5\x00\x00\x00three/b.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd1\xe5v@\x11\x00\x00\x00\v\x00\x00\x00\a\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x14\x01\x00\x00two.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x05\x00\x05\x00>\x01\x00\x00c\x01\x00\x00\x00\x00"}
        	            	actual  : []string{"PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\b\x00\t\x00one%.txtUT\x05\x00\x01%\xc0n8\x00\x05\x00\xfa\xffone%", "\x03\x00PK\a\b\x83\xa7\x83@\f\x00\x00\x00\x05\x00\x00\x00PK\x03\x04\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00three/UT\x05\x00\x01%\xc0n8PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/a.txtUT\x05\x00\x01%\xc0n8\x00\x06\x00\xf9\xffthree", "\x03\x00PK\a\b\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/b.txtUT\x05\x00\x01%\xc0n8\x00\a\x00\xf8\xffthreeb", "\x03\x00PK\a\b\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\a\x00\t\x00two.txtUT\x05\x00\x01%\xc0n8\x00\v\x00\xf4\xff0123456789", "\x03\x00PK\a\b\xd1\xe5v@\x12\x00\x00\x00\v\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x83\xa7\x83@\f\x00\x00\x00\x05\x00\x00\x00\b\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00one%.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00K\x00\x00\x00three/UT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00x\x00\x00\x00three/a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xc7\x00\x00\x00three/b.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd1\xe5v@\x12\x00\x00\x00\v\x00\x00\x00\a\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x17\x01\x00\x00two.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x05\x00\x05\x00>\x01\x00\x00g\x01\x00\x00\x00\x00"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,7 @@
        	            	-([]string) (len=1) {
        	            	- (string) (len=695) "PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\b\x00\t\x00one%.txtUT\x05\x00\x01%\xc0n8\xca\xcfKU\xe5\x02\x04\x00\x00\xff\xffPK\a\b\x83\xa7\x83@\v\x00\x00\x00\x05\x00\x00\x00PK\x03\x04\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00three/UT\x05\x00\x01%\xc0n8PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/a.txtUT\x05\x00\x01%\xc0n8*\xc9(JM\xe5\x02\x04\x00\x00\xff\xffPK\a\b\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/b.txtUT\x05\x00\x01%\xc0n8*\xc9(JMM\xe2\x02\x04\x00\x00\xff\xffPK\a\b\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\a\x00\t\x00two.txtUT\x05\x00\x01%\xc0n820426153\xb7\xb0\xe4\x02\x04\x00\x00\xff\xffPK\a\b\xd1\xe5v@\x11\x00\x00\x00\v\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x83\xa7\x83@\v\x00\x00\x00\x05\x00\x00\x00\b\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00one%.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00J\x00\x00\x00three/UT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00w\x00\x00\x00three/a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xc5\x00\x00\x00three/b.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd1\xe5v@\x11\x00\x00\x00\v\x00\x00\x00\a\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x14\x01\x00\x00two.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x05\x00\x05\x00>\x01\x00\x00c\x01\x00\x00\x00\x00"
        	            	+([]string) (len=5) {
        	            	+ (string) (len=56) "PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\b\x00\t\x00one%.txtUT\x05\x00\x01%\xc0n8\x00\x05\x00\xfa\xffone%",
        	            	+ (string) (len=123) "\x03\x00PK\a\b\x83\xa7\x83@\f\x00\x00\x00\x05\x00\x00\x00PK\x03\x04\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00three/UT\x05\x00\x01%\xc0n8PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/a.txtUT\x05\x00\x01%\xc0n8\x00\x06\x00\xf9\xffthree",
        	            	+ (string) (len=79) "\x03\x00PK\a\b\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\v\x00\t\x00three/b.txtUT\x05\x00\x01%\xc0n8\x00\a\x00\xf8\xffthreeb",
        	            	+ (string) (len=79) "\x03\x00PK\a\b\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\a\x00\t\x00two.txtUT\x05\x00\x01%\xc0n8\x00\v\x00\xf4\xff0123456789",
        	            	+ (string) (len=358) "\x03\x00PK\a\b\xd1\xe5v@\x12\x00\x00\x00\v\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x83\xa7\x83@\f\x00\x00\x00\x05\x00\x00\x00\b\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00one%.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\x00\x00\x00\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00K\x00\x00\x00three/UT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00x\x00\x00\x00three/a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00\v\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xc7\x00\x00\x00three/b.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd1\xe5v@\x12\x00\x00\x00\v\x00\x00\x00\a\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x17\x01\x00\x00two.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x05\x00\x05\x00>\x01\x00\x00g\x01\x00\x00\x00\x00"
        	            	 }
        	Test:       	TestAuthProxy
        	Messages:   	testdata/golden/root.zip
    http_test.go:107: 
        	Error Trace:	/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:107
        	            				/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:289
        	            				/builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/http_test.go:298
        	Error:      	Not equal: 
        	            	expected: []string{"PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00a.txtUT\x05\x00\x01%\xc0n8*\xc9(JM\xe5\x02\x04\x00\x00\xff\xffPK\a\b\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00b.txtUT\x05\x00\x01%\xc0n8*\xc9(JMM\xe2\x02\x04\x00\x00\xff\xffPK\a\b\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00H\x00\x00\x00b.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x02\x00\x02\x00x\x00\x00\x00\x91\x00\x00\x00\x00\x00"}
        	            	actual  : []string{"PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00a.txtUT\x05\x00\x01%\xc0n8\x00\x06\x00\xf9\xffthree", "\x03\x00PK\a\b\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00b.txtUT\x05\x00\x01%\xc0n8\x00\a\x00\xf8\xffthreeb", "\x03\x00PK\a\b\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00I\x00\x00\x00b.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x02\x00\x02\x00x\x00\x00\x00\x93\x00\x00\x00\x00\x00"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,5 @@
        	            	-([]string) (len=1) {
        	            	- (string) (len=287) "PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00a.txtUT\x05\x00\x01%\xc0n8*\xc9(JM\xe5\x02\x04\x00\x00\xff\xffPK\a\b\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00b.txtUT\x05\x00\x01%\xc0n8*\xc9(JMM\xe2\x02\x04\x00\x00\xff\xffPK\a\b\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\f\x00\x00\x00\x06\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\r\x00\x00\x00\a\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00H\x00\x00\x00b.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x02\x00\x02\x00x\x00\x00\x00\x91\x00\x00\x00\x00\x00"
        	            	+([]string) (len=3) {
        	            	+ (string) (len=54) "PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00a.txtUT\x05\x00\x01%\xc0n8\x00\x06\x00\xf9\xffthree",
        	            	+ (string) (len=73) "\x03\x00PK\a\b\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00PK\x03\x04\x14\x00\b\x00\b\x00\x82\x18\"(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\t\x00b.txtUT\x05\x00\x01%\xc0n8\x00\a\x00\xf8\xffthreeb",
        	            	+ (string) (len=160) "\x03\x00PK\a\b\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\xd8\xc5F\xff\r\x00\x00\x00\x06\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00a.txtUT\x05\x00\x01%\xc0n8PK\x01\x02\x14\x00\x14\x00\b\x00\b\x00\x82\x18\"(\x1f\xfb\x04\x17\x0e\x00\x00\x00\a\x00\x00\x00\x05\x00\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00I\x00\x00\x00b.txtUT\x05\x00\x01%\xc0n8PK\x05\x06\x00\x00\x00\x00\x02\x00\x02\x00x\x00\x00\x00\x93\x00\x00\x00\x00\x00"
        	            	 }
        	Test:       	TestAuthProxy
        	Messages:   	testdata/golden/three.zip
2026/07/13 20:56:59 NOTICE: Local file system at /builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/testdata/files: HTTP Server started on [http://127.0.0.1:44765/]
2026/07/13 20:56:59 NOTICE: Local file system at /tmp/TestFaviconremote_override3527098889/001: HTTP Server started on [http://127.0.0.1:46507/]
2026/07/13 20:56:59 NOTICE: Local file system at /builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/testdata/files: HTTP Server started on [http://127.0.0.1:45201/]
2026/07/13 20:56:59 NOTICE: Local file system at /builddir/build/BUILD/rclone-1.74.3-build/rclone-1.74.3/_build/src/github.com/rclone/rclone/cmd/serve/http/testdata/files: HTTP Server started on [http://127.0.0.1:37131/]
2026/07/13 20:56:59 NOTICE: Local file system at /tmp/TestRc3667380901/001: HTTP Server started on [http://127.0.0.1:42655/]
FAIL
~~~

Problem seems to be related to Go 1.27 changes in compress/flate output, which also changes archive/zip
byte output.

### rclone version

rclone 1.74.4 & master
Fedora Rawhide
Golang 1.27rc2

### Operating system

Linux

### Which cloud storage system are you using?

none

### Output of `rclone config redacted`

```ini
n/a
```

### The command you were trying to run

```shell
n/a
```

### A log from the command with the `-vv` flag

```shell
n/a
```

### Anything else?

_No response_

--- a/cmd/serve/http/http_test.go
+++ b/cmd/serve/http/http_test.go
@@ -1,6 +1,8 @@
 package http
 
 import (
+	"archive/zip"
+	"bytes"
 	"compress/gzip"
 	"context"
 	"flag"
@@ -24,6 +26,11 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
+
+type zipEntry struct {
+	IsDir    bool
+	Contents string
+}
 
 var (
 	updateGolden = flag.Bool("updategolden", false, "update golden files for regression test")
@@ -102,10 +109,37 @@ func checkGolden(t *testing.T, fileName string, got []byte) {
 	} else {
 		want, err := os.ReadFile(fileName)
 		require.NoError(t, err)
+		if strings.HasSuffix(fileName, ".zip") {
+			assert.Equal(t, readZip(t, want), readZip(t, got), fileName)
+			return
+		}
 		wants := strings.Split(string(want), "\n")
 		gots := strings.Split(string(got), "\n")
 		assert.Equal(t, wants, gots, fileName)
 	}
+}
+
+func readZip(t *testing.T, data []byte) map[string]zipEntry {
+	t.Helper()
+	zr, err := zip.NewReader(bytes.NewReader(data), int64(len(data)))
+	require.NoError(t, err)
+
+	entries := make(map[string]zipEntry, len(zr.File))
+	for _, f := range zr.File {
+		entry := zipEntry{
+			IsDir: f.FileInfo().IsDir(),
+		}
+		if !entry.IsDir {
+			rc, err := f.Open()
+			require.NoError(t, err)
+			contents, err := io.ReadAll(rc)
+			require.NoError(t, err)
+			require.NoError(t, rc.Close())
+			entry.Contents = string(contents)
+		}
+		entries[f.Name] = entry
+	}
+	return entries
 }
 
 func testGET(t *testing.T, useProxy bool) {

